### ISSUE TITLE
I cant create a new case

### ISSUE THREAD

**Issue by Amanlte** [judsonmitchell/ClinicCases#4]
When i click on New Case i cant get the form to fill out the new case data. Here is below the Error.
casesMessages.js:116 Uncaught TypeError: Cannot read property 'error' of null(anonymous function) @ casesMessages.js:116handleSuccess @ jquery-1.4.4.min.js:142w.onreadystatechange @ jquery-1.4.4.min.js:141
I installed ClinicCases in Centos httpd 2.4.6, in case you need these details.

Please help me with it ASAP.

**Comment by judsonmitchell** [comment]
I don't think this error is caused by casesMessages.js. To debug, have a look in Developer Tools at the network requests when you press "New Case". See what the response is to the request to create_new_case.php and cases_case_data_load.php. That should give you a clue to the problem; usually it is a database configuration error.

**Comment by jameskraus** [comment]
Also got this problem. Seems to be a problem with a missing default value.

**Comment by judsonmitchell** [comment]
This is a result of mysql strict mode. Newer Ubuntu distributions are shipping with mysql strict mode enabled by default, and I'm sure that's now the case with other distros.  I plan to review the db schema and queries to make it compliant with strict mode, but that may take a while. (Happy for anybody to help with that!). In the meantime, you can either disable strict mode in your mysql config or, alternatively, replace the db.php in the ClinicCases distribution with this:

```

<?php
require_once('_CONFIG.php');
try {
    $dbh = new PDO("mysql:host=" . CC_DBHOST . ";dbname=" . CC_DATABASE_NAME , CC_DBUSERNAME, CC_DBPASSWD, array(PDO::MYSQL_ATTR_INIT_COMMAND => 'SET sql_mode="IGNORE_SPACE,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"') );
 }
catch(PDOException $e) {
    //400 is sent to trigger an error for ajax requests.
    header('HTTP/1.1 400 Bad Request');
    echo $e->getMessage();
 }

```

Let me know how that goes.

**Comment by arnisjuraga** [comment]
I have just started to use ClinicCases with MariaDB 10.
Three main reasons for errors:
- no `Default` value in DB scheme fields 
- incorrect (empty, NULL) `datetime` 
- Incorrect string value (when UTF-8 characters used).

I will create a fork of the project to maintain all updates.

**Comment by fhadad** [comment]
> 
> <?php
> require_once('_CONFIG.php');
> try {
>     $dbh = new PDO("mysql:host=" . CC_DBHOST . ";dbname=" . CC_DATABASE_NAME , CC_DBUSERNAME, CC_DBPASSWD, array(PDO::MYSQL_ATTR_INIT_COMMAND => 'SET sql_mode="IGNORE_SPACE,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"') );
>  }
> catch(PDOException $e) {
>     //400 is sent to trigger an error for ajax requests.
>     header('HTTP/1.1 400 Bad Request');
>     echo $e->getMessage();
>  }

This solve the issue for me. Thank you! 👍

**Comment by fhadad** [comment]
> 
> 
> This is a result of mysql strict mode. Newer Ubuntu distributions are shipping with mysql strict mode enabled by default, and I'm sure that's now the case with other distros. I plan to review the db schema and queries to make it compliant with strict mode, but that may take a while. (Happy for anybody to help with that!). In the meantime, you can either disable strict mode in your mysql config or, alternatively, replace the db.php in the ClinicCases distribution with this:
> 
> ```
> 
> <?php
> require_once('_CONFIG.php');
> try {
>     $dbh = new PDO("mysql:host=" . CC_DBHOST . ";dbname=" . CC_DATABASE_NAME , CC_DBUSERNAME, CC_DBPASSWD, array(PDO::MYSQL_ATTR_INIT_COMMAND => 'SET sql_mode="IGNORE_SPACE,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"') );
>  }
> catch(PDOException $e) {
>     //400 is sent to trigger an error for ajax requests.
>     header('HTTP/1.1 400 Bad Request');
>     echo $e->getMessage();
>  }
> ```
> 
> Let me know how that goes.

This solved the issue for me. Thank you! 👍
